File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.py
def solution(record):
    answer = []
    user_dic = {}
    for data in record :
        line = data.split()
        if(line[0]=='Change' or line[0]=='Enter') :
            user_dic[line[1]] = line[2]
            
    for data in record :
        line = data.split()
        if(line[0]=='Enter'):
            answer.append(user_dic[line[1]] + '님이 들어왔습니다.')
        if(line[0]=='Leave'):
            answer.append(user_dic[line[1]] + '님이 나갔습니다.')
            
    return answer

File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.js
function solution(record) {
    var answer = [];
    const user = {}
    for(let r of record){
        let [act,id,name] = r.split(' ');
        if(name) user[id] = name;
    }
    for(let r of record){
        let [act,id,name] = r.split(' ');
        if(act!=='Change') answer.push(`${user[id]}님이 ${act==='Enter'?'들어왔습니다.':'나갔습니다.'}`)
    }
    
    return answer;
}